Packages/com.unity.inputsystem/Tests/TestFixture/InputTestFixture.cs

```diff
@@ -838,27 +838,27 @@ private bool Verify(InputActionTrace.ActionEventPtr eventPtr)
                 if (value != null)
                 {
                     var val = eventPtr.ReadValueAsObject();
-                    if (value is float f)
+                    if (val is float f)
                     {
-                        if (!Mathf.Approximately(f, Convert.ToSingle(val)))
+                        if (!Mathf.Approximately(f, Convert.ToSingle(value)))
                             return false;
                     }
-                    else if (value is double d)
+                    else if (val is double d)
                     {
-                        if (!Mathf.Approximately((float)d, (float)Convert.ToDouble(val)))
+                        if (!Mathf.Approximately((float)d, (float)Convert.ToDouble(value)))
                             return false;
                     }
-                    else if (value is Vector2 v2)
+                    else if (val is Vector2 v2)
                     {
-                        if (!Vector2EqualityComparer.Instance.Equals(v2, val.As<Vector2>()))
+                        if (!Vector2EqualityComparer.Instance.Equals(v2, value.As<Vector2>()))
                             return false;
                     }
-                    else if (value is Vector3 v3)
+                    else if (val is Vector3 v3)
                     {
-                        if (!Vector3EqualityComparer.Instance.Equals(v3, val.As<Vector3>()))
+                        if (!Vector3EqualityComparer.Instance.Equals(v3, value.As<Vector3>()))
                             return false;
                     }
-                    else if (!value.Equals(val))
+                    else if (!val.Equals(value))
```

Comment:
This is a minor tweak to the framework. Previously we used the passed in value to determine the value type of the comparison. It makes more sense to let the actual control type determine the value type so I went and swapped the roles.

```
// Before: Fails if the control happens to be a float control.
// Now: 0 will automatically be cast to a float because the control is a float control.
Assert.That(trace, Canceled(action, value: 0);
```